Source/Core/Duality/Serialization/Serializer.cs

```diff
@@ -317,10 +317,16 @@ protected virtual void OnDisposed(bool manually)
 		public object ReadObject()
 		{
 			if (!this.CanRead) throw new InvalidOperationException("Can't read object from a write-only serializer!");
-			this.BeginReadOperation();
-			object result = this.ReadObjectData();
-			this.EndReadOperation();
-			return result;
+
+			try
+			{
+				this.BeginReadOperation();
+				return this.ReadObjectData();
```

Comment:
When a exception occurred here the serializer would never call EndReadOperation. This in turn caused another exception when the serializer was disposed which would hide the real error. Might wanna move this fix to a separate PR or just leave it in here as thats easier. I put it in a separate commit just in case.

Reply:
I'd be fine with leaving it in this PR, since it's in a separate commit 👍 Just have to remember not to squash-merge this one 🙂